### PR TITLE
Implements pluggable arb annotation handlers

### PR DESCRIPTION
**Pull request by fyoshimatsu**
ARB annotations may have to be handled in different ways by callers. This PR allows callers to specify functions for a particular ARB annotation to allow callers to decide what to do for each annotation.

helpers-json/I18nextFilter accepts the new parameter `arbAnnotationHandlers` whose value is an object of

```
{
  <name of ARB annotation>: (name, data) => <function that returns a string or undefined>,
  ...
}
```

The intention is so we can replace the current `placeholders` handler that simply does `JSON.stringify` with something that replaces them with `PH(...)`.

### CODE REVIEW

File: helpers-json/utils.js [hunk 1]
```diff
@@ -81,8 +81,34 @@ function flattenAndSplitResources(keys, obj) {
...
+ *
+ * ```
+ * const placeholders = {
+ *   count: {
+ *     example: "1",
+ *     description: "number of tickets"
+ *   }
+ * }
+ * const ph = arbPlaceholderHandler(placeholders)
+ * assert(ph === "PH(count|1|number of tickets")
```
Comment (theresakhoo) [inline helpers-json/utils.js:95]: Wouldn't this be

> "PH({{count}}|1|number of tickets)"

Comment (fyoshimatsu) [reply helpers-json/utils.js]: done

File: helpers-json/utils.js [hunk 2]
```diff
@@ -92,7 +92,7 @@ function flattenAndSplitResources(keys, obj) {
  *   }
  * }
  * const ph = arbPlaceholderHandler(placeholders)
- * assert(ph === "PH(count|1|number of tickets")
+ * assert(ph === "PH({{count}}|1|number of tickets")
```
Comment (theresakhoo) [inline helpers-json/utils.js:95]: Nit: there should be a ")" after tickets

Comment (fyoshimatsu) [reply helpers-json/utils.js]: done

**Pull request merged**